sqlglot/dialects/duckdb.py

```diff
@@ -89,6 +89,11 @@ def _struct_sql(self: generator.Generator, expression: exp.Struct) -> str:
 def _datatype_sql(self: generator.Generator, expression: exp.DataType) -> str:
     if expression.is_type("array"):
         return f"{self.expressions(expression, flat=True)}[]"
+
+    # Type TIMESTAMP WITH TIME ZONE does not support any modifiers
+    if expression.is_type("timestamptz", "timetz"):
+        return "TIMESTAMPTZ"
+
```

Comment:
This changed was based on an error I saw while testing:

```python
>>> import duckdb
>>> duckdb.connect().execute("select cast('2020-01-01 00:00:00' as timestamptz(6))").fetchdf()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
duckdb.ParserException: Parser Error: Type TIMESTAMP WITH TIME ZONE does not support any modifiers!
```

Note that DuckDB has neither a `timetz` type, nor a `time with time zone` one, so the `timetz` check here is a temporary solution to transpiling such types from e.g. presto to it. I think in the future we might want to implement a different workaround for this to allow single "time" values with timezone information.